### PR TITLE
chore: Fix linter findings for makezero (part4)

### PR DESCRIPTION
**Pull request by zak-pawel**
Address findings for [makezero](https://github.com/ashanbrown/makezero) - Finds slice declarations with non-zero initial length.

This is the last part of addressing findings found by this linter, thus linter was also enabled in this PR.
related https://github.com/influxdata/telegraf/pull/12254 https://github.com/influxdata/telegraf/pull/12362 https://github.com/influxdata/telegraf/pull/12371

### CODE REVIEW

**Review by powersj — APPROVED** [review]

**Review by srebhan — COMMENTED** [review]
Thanks for the epic effort @zak-pawel! I think I found one issue with your PR, other than that it looks good...

File: plugins/serializers/influx/reader_test.go [hunk 1]
```diff
@@ -235,9 +235,9 @@ func BenchmarkReader(b *testing.B) {
 		},
 		time.Unix(0, 1517620624000000000),
 	)
-	metrics := make([]telegraf.Metric, 1000)
-	for i := range metrics {
-		metrics[i] = m
+	metrics := make([]telegraf.Metric, 0, 1000)
+	for range metrics {
+		metrics = append(metrics, m)
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/serializers/influx/reader_test.go]: @srebhan Great catch! Fixed.

**Review by srebhan — APPROVED** [review]
LGTM. Thanks for the update @zak-pawel!

**Comment by Hipska** [comment]
Is it actually better to do `array = append(array, item)` instead of `array[i] = item`? It looks to me the latter one is more efficient?

**Pull request merged**